crates/dpapi-pdu/src/rpc/epm.rs

```diff
@@ -677,18 +749,42 @@ impl DecodeOwned for EptMapResult {
             .map(|_| {
                 ensure_size!(in: src, size: 8 /* tower length */ + 4 + 2 /* floor length */);
 
-                let tower_length = src.read_u64();
-
+                let tower_length = { cast_length!("EptMap", "tower length", src.read_u64()) as DecodeResult<_> }?;
+                if tower_length < 2
+                /* floor length */
+                {
+                    return Err(DecodeError::invalid_field(
+                        "EptMap",
+                        "tower length",
+                        "tower length is too small",
+                    ));
+                }
+
+                // encoded tower length
                 src.read_u32();
 
+                let tower_start = src.pos();
                 let floor_length = src.read_u16();
                 let tower = (0..floor_length)
                     .map(|_| Floor::decode_owned(src))
                     .collect::<DecodeResult<Vec<Floor>>>()?;
 
+                // invalid tower_length can lead to invalid padding and corrupted fields.
```

Comment:
style: Proper sentence for comments:
```suggestion
                // Invalid tower_length can lead to invalid padding and corrupted fields.
```